### PR TITLE
Add HPCG benchmark

### PR DESCRIPTION
**Pull request by gareth-ferneyhough**
* Add CUDA-enabled [HPCG benchmark](http://www.hpcg-benchmark.org/)
* Add `gpu_autoboost_enabled` flag to cuda_toolkit 
* Add method in cuda_toolkit to get metadata associated with GPU-specific flags

### CODE REVIEW

File: perfkitbenchmarker/data/run_hpcg [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (dlott) [inline perfkitbenchmarker/data/run_hpcg]: run_hpcg.sh?

File: perfkitbenchmarker/data/run_hpcg [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+HOSTNAME=`hostname`
+DATETIME=`hostname`.`date +"%m%d.%H%M%S"`
+
+MPIFLAGS="--mca btl tcp,sm,self"   # just to get rid of warning on psg cluster node wo proper IB sw installed
+
+
+HPCG_BIN=xhpcg-3.1_gcc_485_cuda8061_ompi_1_10_2_sm_35_sm_50_sm_60_ver_3_28_17
+#HPCG_BIN=xhpcg-3.1_gcc_485_cuda8061_ompi_1_6_5_sm_35_sm_50_sm_60_ver_3_28_17
```
Comment (dlott) [inline perfkitbenchmarker/data/run_hpcg]: why might we need this? or should it be deleted?

File: perfkitbenchmarker/linux_benchmarks/hpcg_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run HPCG.
+
+Requires:
+* openmpi 1.10.2 or openmpi 1.6.5
+
+Binaries are provided which are statically linked to CUDA 8.0.44 and
```
Comment (dlott) [inline perfkitbenchmarker/linux_benchmarks/hpcg_benchmark.py]: why not pick one?

File: perfkitbenchmarker/linux_benchmarks/hpcg_benchmark.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+      vm_count: null
+"""
+
+flags.DEFINE_integer(
+    'hpcg_runtime', 60, 'hpcg runtime in seconds', lower_bound=1)
+
+flags.DEFINE_integer(
+    'hpcg_gpus_per_node', None, 'The number of gpus per node.', lower_bound=1)
+
+flag_util.DEFINE_integerlist(
```
Comment (dlott) [inline perfkitbenchmarker/linux_benchmarks/hpcg_benchmark.py]: how does this interact with hpcg.dat?

Comment (dlott) [inline perfkitbenchmarker/linux_benchmarks/hpcg_benchmark.py]: since it has a default, will the values in hpcg.dat ever be used?

Comment (gareth-ferneyhough) [inline perfkitbenchmarker/linux_benchmarks/hpcg_benchmark.py]: This is used to set the number of mpi processes in `run_hpcg.sh`. `hpcg.dat` has no effect on the number of gpus used.

File: perfkitbenchmarker/data/run_hpcg.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+HPCG_DIR=`pwd`
+
+
+HOSTNAME=`hostname`
```
Comment (dlott) [inline perfkitbenchmarker/data/run_hpcg.sh]: unused?

**Comment by dlott** [comment]
LGTM

**Pull request merged**